src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1290,7 +1311,22 @@ std::unique_ptr<StorageTargetRHandle> SQLStorage::openTargetFile(const Uptane::T
 void SQLStorage::removeTargetFile(const std::string& filename) {
   SQLite3Guard db = dbConnection();
 
-  auto statement = db.prepareStatement<std::string>("DELETE FROM target_images WHERE filename=?;", filename);
+  auto statement = db.prepareStatement<std::string>("SELECT image_id FROM target_images WHERE filename = ?;", filename);
+
+  if (statement.step() != SQLITE_ROW) {
+    LOG_ERROR << "Statement step failure: " << db.errmsg();
+    throw std::runtime_error("Could not find target file");
+  }
+  auto image_id = statement.get_result_col_int(0);
+
+  statement = db.prepareStatement<std::string>("DELETE FROM target_images WHERE filename=?;", filename);
+
+  if (statement.step() != SQLITE_DONE) {
+    LOG_ERROR << "Statement step failure: " << db.errmsg();
+    throw std::runtime_error("Could not remove target file");
+  }
+
+  statement = db.prepareStatement<int64_t>("DELETE FROM target_images_data WHERE rowid=?;", image_id);
```

Comment:
Should it be needed? The `ON DELETE CASCADE` clause is supposed to propagate deletion?

Reply:
Cascade delete works only in reverse mode. But Yes we could just delete blob row from 'target_images_data' and appropirate row in target_images will be removed.